src/main.rs

```diff
@@ -288,24 +288,13 @@ pub fn set_var(variables: &mut Variables, name: &str, value: &str) {
     }
 }
 
-fn print_prompt(modes: &Vec<Mode>) {
-    for mode in modes.iter().rev() {
-        if mode.value {
-            print!("+ ");
-        } else {
-            print!("- ");
-        }
-    }
+fn print_prompt(modes: &[Mode]) {
+    let mode = modes.iter().rev().fold(String::new(), |acc, mode| {
+        acc + if mode.value { "+ " } else { "- " }
+    });
+    print!("{}", mode.trim_right());
```

Comment:
I think that was intentional. If you are in an else branch in an else branch it would say:
`- - user@ion#` or something like that.